File: GFG/LinkedList/recrev.py
import ds

def recrev(curr,prev):
    if(curr is None):
        return prev
    net=curr.next
    curr.next=prev
    return recrev(net,curr)

dll=ds.LinkedList()
for i in range(1,11):
    dll.insert(i*10)
    
dll.display()
qaz=recrev(dll.head,None)
while(qaz!=None):
    print(qaz.data,end="->")
    qaz=qaz.next
    
    
    

File: GFG/LinkedList/recrev.cpp
#include<iostream>
#include"../ds.h"
using namespace std;

Node<int>* recrev(Node<int>* curr,Node<int>* prev){
    if(curr==NULL ){
        return prev;
    }
    Node<int>* next=curr->next;
    curr->next=prev;
    return recrev(next,curr);
}

int main(){
    LinkedList<int> dll;
    for(int i=1;i<11;i++){
        dll.insert(i*10);
    }
    dll.display();
    Node<int>*newh=recrev(dll.head,NULL);
    Node<int>* curr=newh;
    while(curr!=NULL ){
        cout<<curr->data<<"->";
        curr=curr->next;
    }
    return 0;

}